Resilient smart-workflow PMV resolution

Smart-Workflow execution fails after installing (14.0.0-SNAPSHOT) from Market 🛒 . 

Reproduce:
- install smart-workflow via market (vsc command)
- stick to defaults and pick 14.0.0-SNAPSHOT
- execute a demo process
- fails: with a cryptic error on SPI resolution (see below)

Cause:
- PMV name is not the expected "smart-workflow" if installed via market -> Use a more resilient approach to verify we find the correct one 🧠 
<img width="1534" height="883" alt="Image" src="https://github.com/user-attachments/assets/09b181d4-476e-4b48-b271-ded343f61ba7" />

error:
```
Caused by: BpmError ivy:error:program:exception
 Error Id: 19DB996A1E233F83
 Context: request=HTTP GET AgentDemo/SupportAgentTools.p.json/start.ivp(2.2.0.0), session=41 Developer, securityContext=~Developer-vsctest5, task=2, application=Developer-vsctest5, requestId=201, executionContext=SYSTEM, pmv=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-demo, client=127.0.0.1, processElement=19856884121ED111-f3
 Process Element: Ivy Agent <19856884121ED111-f3>
 Attributes:
  JavaClassName=com.axonivy.utils.smart.workflow.AgenticProcessCall
	at ch.ivyteam.ivy.bpm.error.BpmErrorBuilder.build(BpmErrorBuilder.java:194)
	at ch.ivyteam.ivy.bpm.engine.restricted.error.BpmErrorConverter.toBpmError(BpmErrorConverter.java:38)
	at ch.ivyteam.ivy.bpm.engine.internal.model.bpmn.BackgroundActivityExecutorAdapter.handleErrorAndTimeout(BackgroundActivityExecutorAdapter.java:69)
	at ch.ivyteam.ivy.bpm.engine.internal.model.bpmn.BackgroundActivityExecutorAdapter.handleBackgroundOperationResult(BackgroundActivityExecutorAdapter.java:57)
	at ch.ivyteam.ivy.bpm.engine.internal.model.bpmn.BackgroundActivityExecutorAdapter.executeInProcessElementRequest(BackgroundActivityExecutorAdapter.java:43)
	at ch.ivyteam.ivy.bpm.engine.internal.model.bpmn.ProcessElementRequestExecutorAdapter.lambda$0(ProcessElementRequestExecutorAdapter.java:39)
	at ch.ivyteam.util.context.AbstractExecutionContext.callInContext(AbstractExecutionContext.java:11)
	at ch.ivyteam.util.context.IExecutionContext.callUncheckedInContext(IExecutionContext.java:33)
	at ch.ivyteam.util.context.IExecutionContext.getInContext(IExecutionContext.java:28)
	at ch.ivyteam.ivy.bpm.engine.internal.model.bpmn.ProcessElementRequestExecutorAdapter.executeInRequest(ProcessElementRequestExecutorAdapter.java:38)
	at ch.ivyteam.ivy.bpm.engine.internal.model.bpmn.ProcessElementRequestExecutorAdapter.execute(ProcessElementRequestExecutorAdapter.java:32)
	at ch.ivyteam.ivy.bpm.engine.internal.model.ProcessElement.lambda$1(ProcessElement.java:218)
	at ch.ivyteam.util.context.AbstractExecutionContext.callInContext(AbstractExecutionContext.java:11)
	at ch.ivyteam.util.context.IExecutionContext.callUncheckedInContext(IExecutionContext.java:33)
	at ch.ivyteam.util.context.IExecutionContext.getInContext(IExecutionContext.java:28)
	at ch.ivyteam.ivy.bpm.engine.internal.model.ProcessElement.executeInElementContext(ProcessElement.java:273)
	at ch.ivyteam.ivy.bpm.engine.internal.model.ProcessElement.execute(ProcessElement.java:218)
	at ch.ivyteam.ivy.bpm.engine.internal.background.FinishBackgroundOperation.execute(FinishBackgroundOperation.java:33)
	at ch.ivyteam.ivy.bpm.engine.internal.core.RequestProcessor.handleExecution(RequestProcessor.java:125)
	at ch.ivyteam.ivy.bpm.engine.internal.core.RequestProcessor.processRequest(RequestProcessor.java:117)
	at ch.ivyteam.ivy.bpm.engine.internal.core.RequestProcessor.handleRequest(RequestProcessor.java:89)
	at ch.ivyteam.ivy.bpm.engine.internal.core.RequestProcessor.handleRequest(RequestProcessor.java:72)
	at ch.ivyteam.ivy.bpm.engine.internal.BpmEngine.handleNormalRequest(BpmEngine.java:157)
	... 147 more
Caused by: java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.orElseThrow(Optional.java:377)
	at IvyProjectClassLoader [project=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-14-0-0-SNAPSHOT,generation=3]//com.axonivy.utils.smart.workflow.spi.internal.SpiProject.getSmartWorkflowPmv(SpiProject.java:17)
	at IvyProjectClassLoader [project=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-14-0-0-SNAPSHOT,generation=3]//com.axonivy.utils.smart.workflow.model.ChatModelFactory.providers(ChatModelFactory.java:44)
	at IvyProjectClassLoader [project=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-14-0-0-SNAPSHOT,generation=3]//com.axonivy.utils.smart.workflow.model.ChatModelFactory.create(ChatModelFactory.java:38)
	at IvyProjectClassLoader [project=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-14-0-0-SNAPSHOT,generation=3]//com.axonivy.utils.smart.workflow.model.ChatModelFactory.getProviderOrDefault(ChatModelFactory.java:33)
	at IvyProjectClassLoader [project=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-14-0-0-SNAPSHOT,generation=3]//com.axonivy.utils.smart.workflow.program.internal.AgentCallExecutor.configureModel(AgentCallExecutor.java:124)
	at IvyProjectClassLoader [project=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-14-0-0-SNAPSHOT,generation=3]//com.axonivy.utils.smart.workflow.program.internal.AgentCallExecutor.execute(AgentCallExecutor.java:68)
	at IvyProjectClassLoader [project=~Developer-vsctest5/Developer-vsctest5/1/smart-workflow-14-0-0-SNAPSHOT,generation=3]//com.axonivy.utils.smart.workflow.AgenticProcessCall.lambda$0(AgenticProcessCall.java:18)
	at ch.ivyteam.ivy.bpm.exec.internal.activity.program.executor.ProgramExecBeanHolder.lambda$1(ProgramExecBeanHolder.java:69)
	at ch.ivyteam.util.context.IExecutionContext.lambda$0(IExecutionContext.java:15)
	at ch.ivyteam.util.context.AbstractExecutionContext.callInContext(AbstractExecutionContext.java:11)
	at ch.ivyteam.util.context.IExecutionContext.executeInContext(IExecutionContext.java:14)
	at ch.ivyteam.ivy.bpm.exec.internal.activity.program.executor.ProgramExecBeanHolder.run(ProgramExecBeanHolder.java:66)
	at ch.ivyteam.ivy.bpm.exec.internal.activity.program.executor.ProgramBackgroundOperation.execute(ProgramBackgroundOperation.java:26)
	at ch.ivyteam.ivy.bpm.exec.internal.activity.program.executor.ProgramBackgroundOperation.execute(ProgramBackgroundOperation.java:1)
	at ch.ivyteam.ivy.bpm.engine.internal.background.wrapper.BackgroundOperationWrapper.execute(BackgroundOperationWrapper.java:21)
	at ch.ivyteam.ivy.bpm.engine.internal.background.wrapper.ExecuteBackgroundOperationAsSessionUserWrapper.access$0(ExecuteBackgroundOperationAsSessionUserWrapper.java:1)
	at ch.ivyteam.util.context.AbstractExecutionContext.callInContext(AbstractExecutionContext.java:11)
	at ch.ivyteam.ivy.security.internal.context.SecurityContext.executeAs(SecurityContext.java:218)
	at ch.ivyteam.ivy.bpm.engine.internal.background.wrapper.ExecuteBackgroundOperationAsSessionUserWrapper.execute(ExecuteBackgroundOperationAsSessionUserWrapper.java:29)
	at ch.ivyteam.ivy.bpm.engine.internal.background.wrapper.BackgroundOperationWrapper.execute(BackgroundOperationWrapper.java:21)
	at ch.ivyteam.ivy.bpm.engine.internal.background.wrapper.ExecuteBackgroundOperationInRequestThreadContextWrapper.execute(ExecuteBackgroundOperationInRequestThreadContextWrapper.java:22)
	at ch.ivyteam.ivy.bpm.engine.internal.background.BackgroundTask.call(BackgroundTask.java:37)
	at ch.ivyteam.ivy.bpm.engine.internal.background.BackgroundTask.call(BackgroundTask.java:1)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:545)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	... 1 more
```

--- a/smart-workflow/src/com/axonivy/utils/smart/workflow/spi/internal/SpiProject.java
+++ b/smart-workflow/src/com/axonivy/utils/smart/workflow/spi/internal/SpiProject.java
@@ -5,15 +5,23 @@ import java.util.function.Predicate;
 import ch.ivyteam.ivy.application.IProcessModelVersion;
 
 public final class SpiProject {
-  static String SMART_WORKFLOW_PROJECT = "smart-workflow";
+
+  private interface SmartWorkflow {
+    String GROUP_ID = "com.axonivy.utils.ai";
+    String ARTIFACT_ID = "smart-workflow";
+    String LIBRARY_ID = GROUP_ID + ":" + ARTIFACT_ID;
+  }
 
   public static IProcessModelVersion getSmartWorkflowPmv() {
-    Predicate<IProcessModelVersion> smartWorkflow = pmv -> SMART_WORKFLOW_PROJECT.equals(pmv.getName());
+    Predicate<IProcessModelVersion> smartWorkflow = pmv -> 
+      SmartWorkflow.LIBRARY_ID.equals(pmv.getLibraryId());
     var current = IProcessModelVersion.current();
     if (smartWorkflow.test(current)) {
       return current;
     }
     return current.getAllRequiredProcessModelVersions()
-        .filter(smartWorkflow).findAny().orElseThrow();
+        .filter(smartWorkflow)
+        .findAny()
+        .orElseThrow();
   }
 }

--- a/smart-workflow/src/com/axonivy/utils/smart/workflow/spi/internal/SpiProject.java
+++ b/smart-workflow/src/com/axonivy/utils/smart/workflow/spi/internal/SpiProject.java
@@ -20,8 +20,12 @@ public final class SpiProject {
       return current;
     }
     return current.getAllRequiredProcessModelVersions()
+        .toList().stream()
         .filter(smartWorkflow)
         .findAny()
-        .orElseThrow();
+        .orElseThrow(() -> new IllegalStateException(String.format(
+            "Cannot resolve Smart Workflow PMV. Expected libraryId='%s', currentLibraryId='%s'",
+            SmartWorkflow.LIBRARY_ID,
+            current.getLibraryId())));
   }
 }

